sentry_sdk/integrations/django/caching.py

```diff
@@ -0,0 +1,101 @@
+import functools
+from typing import TYPE_CHECKING
+
+from django import VERSION as DJANGO_VERSION
+from django.core.cache import CacheHandler
+
+from sentry_sdk import Hub
+from sentry_sdk.consts import OP, SPANDATA
+from sentry_sdk._compat import text_type
+
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Callable
+
+
+METHODS_TO_INSTRUMENT = [
+    "get",
+    "get_many",
+]
+
+
+def _instrument_call(cache, method_name, original_method, args, kwargs):
+    # type: (CacheHandler, str, Callable[..., Any], Any, Any) -> Any
+    description = "{} {}".format(method_name, " ".join(args))
+
+    with Hub.current.start_span(op=OP.CACHE, description=description) as span:
+        cache._sentry_recording = True
```

Comment:
this is stored on the `cache` object so if multiple threads are doing cache stuff, I don't think this works?

 also why is this check necessary? is it because `get_many` calls `get` for some types of caches?

Reply:
Yes, this is done because there is a `get_or_set` function that first calls `get` and then `set` and I only want to record the cache hit/miss of functions the user actually called and that where not called by other functions.

Reply:
About the thread safety. You are probably right. Did not think about threading. I probably can add a `with treading.lock()` or something to make sure it is thread save. (Will first fix all the other stuff than think about threading)

Reply:
Ok, I will just remove the `_sentry_recording` flag. Because the question we want to have answer is "is my caching setup working great?". And for this it is also important that we record hit/misses no matter if run by the user or by a command the user was calling.